packages/teams-js/src/internal/communication.ts

```diff
@@ -257,6 +264,90 @@ export function sendMessageToParentAsync<T>(
   });
 }
 
+type SimpleType = string | number | boolean | null | undefined | SimpleType[];
+
+function isSerializable(arg: unknown): arg is ISerializable {
+  return (
+    arg !== undefined &&
+    arg !== null &&
+    (arg as ISerializable).serialize !== undefined &&
+    typeof (arg as ISerializable).serialize === 'function'
+  );
+}
+
+function serializeItemArray(items: (SimpleType | ISerializable)[]): unknown[] {
+  return items.map((item) => {
+    if (isSerializable(item)) {
+      return item.serialize();
+    } else {
+      return item;
+    }
+  });
+}
+
+/**
+ * Call a function in the host and receive a response. If the host returns an {@link SdkError} instead of a normal response, this function will throw a new Error containing the SdkError's information
+ *
+ * @param apiVersionTag A unique tag used to identify the API version for telemetry purposes.
+ * @param functionName The function name to call in the host.
+ * @param responseHandler When the host responds, this handler will validate and deserialize the response.
+ * @param args A collection of data to pass to the host. This data must be an array of either simple types or objects that implement {@link ISerializable}.
+ * @param errorChecker This optional property can be used to override the default ErrorChecking this function uses to decide whether to throw the host response as a new Error. Specify this is your function needs to do any logic verifying that the object received is an error that goes beyond the logic found in {@link isSdkError}.
+ *
+ * @returns The response received from the host after deserialization.
+ *
+ * @throws An Error containing the SdkError information ({@link SdkError.errorCode} and {@link SdkError.message}) if the host returns an SdkError, or an Error if the response from the host is an unexpected format.
+ */
+export async function callFunctionInHostAndHandleResponse<ReceivedFromHost, DeserializedFromHost>(
+  apiVersionTag: string,
```

Comment:
I guess this is sort of nitpicky but it seems kind of weird that the first parameter to this function is an api version tag instead of, you know, the name of the function you want to call.

Reply:
It seems like a sensible order would be: functionName, args, responseHandler, apiVersionTag, errorChecker, based on the way people might think about "calling" a function.